internal/restconfig/restconfig.go

```diff
@@ -256,121 +320,109 @@ func (rcp *Producer) RunOnSelectedPrefixedContext(prefix string, function PerCon
 	return nil
 }
 
-func (rcp *Producer) AddKubeConfigFlag(cmd *cobra.Command) {
-	cmd.PersistentFlags().StringVar(&rcp.kubeConfig, "kubeconfig", "", "absolute path(s) to the kubeconfig file(s)")
-}
-
-// AddKubeContextMultiFlag adds a "kubeconfig" flag and a "kubecontext" flag that can be specified multiple times (or comma separated).
-func (rcp *Producer) AddKubeContextMultiFlag(cmd *cobra.Command, usage string) {
-	rcp.AddKubeConfigFlag(cmd)
+// RunOnAllContexts runs the given function on all accessible non-prefixed contexts.
+// If the user has explicitly selected one or more contexts, only those contexts are used.
+// All appropriate contexts are processed, and any errors are aggregated.
+// Returns true if at least one context was found.
+func (rcp *Producer) RunOnAllContexts(function PerContextFn, status reporter.Interface) (bool, error) {
+	if rcp.inCluster {
+		return true, runInCluster(function, status)
+	}
 
-	if usage == "" {
-		usage = "comma-separated list of kubeconfig contexts to use, can be specified multiple times.\n" +
-			"If none specified, all contexts referenced by the kubeconfig are used"
+	if rcp.defaultClientConfig.overrides.CurrentContext != "" {
+		// The user has explicitly chosen a context, use that only
+		return true, rcp.RunOnSelectedContext(function, status)
 	}
 
-	cmd.PersistentFlags().StringSliceVar(&rcp.kubeContexts, "kubecontexts", nil, usage)
-}
+	if rcp.defaultClientConfig == nil {
+		return false, nil
+	}
 
-// AddInClusterConfigFlag adds a flag enabling in-cluster configurations for processes running in pods.
-func (rcp *Producer) AddInClusterConfigFlag(cmd *cobra.Command) {
-	cmd.PersistentFlags().BoolVar(&rcp.inCluster, "in-cluster", false, "use the in-cluster configuration to connect to Kubernetes")
-}
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		rcp.defaultClientConfig.loadingRules, rcp.defaultClientConfig.overrides)
 
-func (rcp *Producer) PopulateTestFramework() {
-	framework.TestContext.KubeContexts = rcp.kubeContexts
-	if rcp.kubeConfig != "" {
-		framework.TestContext.KubeConfig = rcp.kubeConfig
+	rawConfig, err := clientConfig.RawConfig()
+	if err != nil {
+		return false, status.Error(err, "error retrieving the raw kubeconfig setup")
 	}
-}
 
-func (rcp *Producer) MustGetForClusters() []RestConfig {
-	configs, err := rcp.getRestConfigs()
-	exit.OnErrorWithMessage(err, "Error getting REST Config for cluster")
+	contextErrors := []error{}
+	processedContexts := 0
 
-	return configs
-}
+	if len(rcp.contexts) > 0 {
+		// Loop over explicitly-chosen contexts
+		for _, contextName := range rcp.contexts {
+			chosenContext, ok := rawConfig.Contexts[contextName]
+			if !ok {
+				status.Failure("error retrieving context %s", contextName)
```

Comment:
Shouldn't we add a `contextError` here? Otherwise it seems it will log a failure message but won't fail it.